app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 # app/controllers/posts_controller.rb
 class PostsController < ApplicationController
+  before_action :set_user, only: %i[index new create]
+
   def index
-    @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:comments).paginate(page: params[:page], per_page: 6)
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.build(post_params)
+    if @post.save
+      redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+    else
+      render :new
+    end
```

Comment:
- [ ] I was unable to create a new post successfully for the `current_user`, kindly work around the `current_user` implementation properly as suggested earlier. 🙏